### PR TITLE
kubeadm template to allow overriding the location.

### PR DESCRIPTION
**Pull request by voor**
This allows you to change where the kubeadm template file is pulled from, this is pretty much leveraged for the `kubeadmpull.yml` task file that preloads container images on the instance.

I keep dirtying my repository due to having a custom kubeadm.yml, now that there's that neat little git version tag addition I wanted to make sure I could override this properly rather than overwriting the file in the repository.

/assign @figo 
/cc @codenrhoden @detiber

### CODE REVIEW

File: images/capi/packer/config/ansible-args.json [hunk 1]
```diff
@@ -1,3 +1,3 @@
...
...ce`}} kubernetes_cni_http_checksum={{user `kubernetes_cni_http_checksum`}} kubernetes_http_source={{user `kubernetes_http_source`}} kubernetes_container_registry={{user `kubernetes_container_registry`}} kubernetes_rpm_repo={{user `kubernetes_rpm_repo`}} kubernetes_rpm_gpg_key={{user `kubernetes_rpm_gpg_key`}} kubernetes_rpm_gpg_check={{user `kubernetes_rpm_gpg_check`}} kubernetes_deb_repo={{user `kubernetes_deb_repo`}} kubernetes_deb_gpg_key={{user `kubernetes_deb_gpg_key`}} kubernetes_cni_deb_version={{user `kubernetes_cni_deb_version`}} kubernetes_cni_rpm_version={{user `kubernetes_cni_rpm_version`}} kubernetes_cni_semver={{user `kubernetes_cni_semver`}} kubernetes_cni_source_type={{user `kubernetes_cni_source_type`}} kubernetes_semver={{user `kubernetes_semver`}} kubernetes_source_type={{user `kubernetes_source_type`}} kubernetes_load_additional_imgs={{user `kubernetes_load_additional_imgs`}} kubernetes_deb_version={{user `kubernetes_deb_version`}} kubernetes_rpm_version={{user `kubernetes_rpm_version`}} no_proxy={{user `no_proxy`}} redhat_epel_rpm={{user `redhat_epel_rpm`}} reenable_public_repos={{user `reenable_public_repos`}} remove_extra_repos={{user `remove_extra_repos`}} "
```
Comment (figo) [inline images/capi/packer/config/ansible-args.json:2]: it's really hard to review this part, is there a better way to generate diff on a long string? just curious.

Comment (dongsupark) [reply images/capi/packer/config/ansible-args.json:2]: > it's really hard to review this part, is there a better way to generate diff on a long string? just curious.

True. It is hard to understand the code and to track down what has changed.
Especially true if a PR has 5 different commits that touch `ansible-args.json`, and you need to rebase the PR every few days. ;-)

Though I have no idea how to simpify this part. A json file does not allow a line break in itself.
Would a yaml file be better? :thinking:

**Pull request merged**